esp/scm/ws_config2.ecm

```diff
@@ -0,0 +1,368 @@
+    bool validateOnly(false);  // only checks validity of new values, no changes are made
+    string localValue("");
+};
+
+
+ESPresponse [exceptions_inline, encode(0)] SetValuesResponse
+{
+    string nodeId;
+    bool   error(false);
+    ESPstruct statusType status;
+};
+
+
+
+ESPresponse [exceptions_inline, encode(0)] GetParentsResponse
+{
+    bool   error(false);
+    string msg("");
+    ESParray<string, parentId> parentIdList;
+};
+
+
+//
+// Environment management
+
+ESPstruct environmentFileType
+{
+    string    filename;
+    bool      isActive(false);  // True if this is the current active environment
+};
+
+
+ESPrequest CommonSessionRequest
+{
+    string sessionId;
+};
+
+
+ESPresponse GetEnvironmentListResponse
+{
+    bool error(false);
+    string msg("");
+    ESParray<ESPstruct environmentFileType, environmentFile> environmentFiles;
+};
+
+
+ESPrequest OpenEnvironmentFileRequest
+{
+    string sessionId;
+    string filename;
+    //string sessionLockKey("");            // required to force open if modified
+    //bool   discardChanges(false);         // discard modifications (only works of current session owns the file)
+    //bool   forceOpenIfModified(false);    // forces any current changes to be discarded
+};
+
+
+ESPrequest CloseEnvironmentFileRequest
+{
+    string sessionId;
+    string sessionLockKey("");            // required if existing environment is modified and discarding changes
+    bool   discardChanges(false);         // discard modifications
+};
+
+
+ESPrequest SaveEnvironmentFileRequest
+{
+    string sessionId;
+    string sessionLockKey("");      // required if saving existing environment and it has been modified
+    string filename("");            // empty string means to overwrite the existing file
+};
+
+
+ESPresponse LockSessionResponse
+{
+    bool error(false);
+    string msg("");
+    string sessionLockKey("");   // filled in when a session is sucessfully locked
+};
+
+
+ESPrequest UnlockSessionRequest
+{
+    string sessionId;
+    string sessionLockKey("");   // only valid for an unlock (enable = false)
+};
+
+
+ESPrequest ValidateEnvironmentRequest
+{
+    string sessionId;
+    bool   includeHiddenNodes(false);   // includes hidden nodes (hidden attributes always included)
+};
+
+
+ESPrequest GetTreeRequest
+{
+    string sessionId;
+    string nodeId;
+    bool   includeAttributes;
+    int    numLevels(1);        // use -1 for all
+};
+
+
+ESPstruct treeElementType
+{
+    string    nodeId;
+    ESPstruct nodeInfoType nodeInfo;
+    ESParray<ESPstruct attributeType, attribute> attributes;
+    ESParray<ESPstruct treeElementType, node> children;
+};
+
+
+ESPresponse GetTreeResponse
+{
+    bool error(false);
+    string msg("");
+    ESPstruct treeElementType tree;
+};
+
+
+
+
+ESPservice [auth_feature("DEFERRED"),version("2.0"), default_client_version("2.0"), exceptions_inline("xslt/exceptions.xslt")] ws_config2
+{
+    ESPMethod openSession(OpenSessionRequest, OpenSessionResponse);
```

Comment:
Capitalize the method names. This will force your ::onOpenSession to be cammelCase

Reply:
Method names capitalized in work for HPCC-19216

Reply:
Method names now capitalized